### PR TITLE
[animations] added onClosed

### PR DESCRIPTION
**Pull request by The-Redhat**
The callback onClosed is called when the open Container has closed.

Fixes [#51458](https://github.com/flutter/flutter/issues/51458)

### CODE REVIEW

**Comment by goderbauer** [comment]
Thanks for the contribution. This PR will need a test before we can accept it.

Also, instead of defining a new onClosed callback, maybe the action callback provided to the closedBuilder should just return the Future from Navigator.push.

File: packages/animations/lib/src/open_container.dart [hunk 1]
```diff
@@ -287,6 +294,7 @@ class _HideableState extends State<_Hideable> {
   /// When non-null the child is replaced by a [SizedBox] of the set size.
   Size get placeholderSize => _placeholderSize;
   Size _placeholderSize;
+
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: Per flutter's style guide there shouldn't be a space here. Please remove.

File: packages/animations/lib/src/open_container.dart [hunk 2]
```diff
@@ -302,6 +310,7 @@ class _HideableState extends State<_Hideable> {
   /// (i.e. [isInTree] returns false).
   bool get isVisible => _visible;
   bool _visible = true;
+
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: Per flutter's style guide there shouldn't be a space here. Please remove.

**Comment by The-Redhat** [comment]
I added a test. I hope it suits your needs. If not I would appreciate some Feedback.

File: packages/animations/test/open_container_test.dart [hunk 3]
```diff
@@ -1448,6 +1448,47 @@ void main() {
...
+    await tester.pumpWidget(
+      _boilerplate(child: openContainer),
+    );
+
+    expect(find.text('Open'), findsNothing);
+    expect(find.text('Closed'), findsOneWidget);
+
+    await tester.tap(find.text('Closed'));
+    await tester.pumpAndSettle();
+
```
Comment (goderbauer) [inline packages/animations/test/open_container_test.dart:1481]: Can you throw an `expect(hasClosed, isFalse);` in here to ensure that the callback doesn't fire too early?

Comment (The-Redhat) [reply packages/animations/test/open_container_test.dart:1482]: added it.

File: packages/animations/lib/src/open_container.dart [hunk 4]
```diff
@@ -163,6 +164,9 @@ class OpenContainer extends StatefulWidget {
   ///  * [Material.shape], which is used to implement this property.
   final ShapeBorder openShape;
 
+  /// Called when the container was popped and has returned to the closed state
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart:167]: nit: end with `.`

**Comment by The-Redhat** [comment]
I rebased the pr.

**Comment by goderbauer** [comment]
@Melvin-Abraham That sounds reasonable. Wanna send a PR for that?

**Comment by Melvin-Abraham** [comment]
@goderbauer I have submitted a PR here: #153. Please do review it :)

**Pull request merged**